resources/types/Genome.yaml

```diff
@@ -1,34 +1,73 @@
+# This specification describes how a Workspace Genome object is transformed into a document that is
```

Comment:
I think the main benefit other than comments is that it's easier for a human to read and write vs. JSON.

Reply:
Also now you can use YAML or JSON as you wish so there's no real drawbacks other than the code to set up and use the YAML library is slightly more complicated than the JSON lib.

Reply:
Also, comments are reason enough with specs this complicated (and versioning is going to make them even more complicated).

Reply:
I totally agree with using YAML. But I don't see the need to support both YAML and JSON. We should just take out JSON for these specs altogether and only support YAML.

Reply:
YAML is a superset* of JSON, so by supporting YAML, we also support JSON.

* There some argument about this, but AFAICT it's very pedantic and has little to no practical impact.

Reply:
If specs written in JSON are not as readable as YAML specs and definitely not documentable, we should not allow devs to write JSON specs. This will force devs to use YAML and write cleaner specs.

Reply:
That's not possible without hacking the yaml library.

Reply:
Just a simple check. If file ends with .json throw exception. Then convert all specs to YAML and the devs will be none the wiser.

Reply:
Well, any dev that knows anything about YAML will know they can put JSON in their YAML specs.

That being said, I don't really care either way, so if you want to switch the specs go for it. The only change I'd make to what you're saying is just skipping the *.json files rather than throwing exceptions.

Reply:
It turns out that with the YAML parser you can put comments in JSON.